### DISCUSSION TITLE
Should tags be case sensitive?

### DISCUSSION THREAD

**Discussion by pde** [11ty/eleventy#1461] [Enhancement Queue]
Moving [an issue](https://github.com/11ty/eleventy-base-blog/issues/71) to discussion here per @zachleat's request.

Currently the tags in the base blog are case sensitive, but I suspect that's not what most users would actually want. In a project I've used them for, I have a mix of tags from different human and programmatic sources, and it's not easy to ensure the human-generated ones are all lower cased.

What would people think about a config option to control this, probably defaulting to "case insensitive"?

**Comment by Ryuno-Ki** [comment]
Another option might be to _convert_ tags to lowercase for display?

**Reply by nhoizey** [reply]
I would even recommend to sluggify tags.

All these tags would have the same `etiquette` slug, the same URL, and "contain" the same collection of items:
- étiquette
- Étiquette
- Etiquette
- etiquette

"étiquette" is the French word for "tag"… 😃

**Reply by Ryuno-Ki** [reply]
That would have been my suggestion once an agreement on lowercasing would appear.
Perhaps also worth looking into  [Levenshtein distance](https://en.wikipedia.org/wiki/Levenshtein_distance) (as plugin) to warn for similar tags.

**Reply by nhoizey** [reply]
I guess Levenshtein distance could be useful in a linter, but similar tags should be able to co-exist.

**Comment by edwardhorsford** [comment]
I've certainly had bugs on my site where i've had duplicate tags as a result of inconsistent captialisation.

I don't know if something should force them to one or the other, but it could _combine_ both cases so that either works.

**Reply by zachleat** [reply]
>  could combine both cases so that either works.

I’m not sure I understand what this means

**Reply by ItaiAxelrad** [reply]
I believe @edwardhorsford  means collections would be merged if they match when case insensitive/lowercase. For example, if you have two tags `posts` and `Posts` the resulting collections would be combined into one and accessed either by `collections.posts` or `collections.Posts`?

**Reply by edwardhorsford** [reply]
Yes, what @ItaiAxelrad describes is what I was thinking

**Comment by johnridesabike** [comment]
It would be great if there was a way to normalize tags, but I think it’s also a bit more complex than this.

For example, I have a site where I would like these following tags to be “merged” into one tag:

```yaml
- Atlanta, GA
- atlanta ga
- Atlanta, Georgia
```
...etc.

What if there was an API for normalizing tags? I can imagine a hypothetical function like this:

```js
eleventyConfig.onCreateTag((tag) => tag.toLowerCase().trim() /* or whatever else you want to do */);
```

Since there’s probably no default configuration that can satisfy everyone, this leaves it in the hands of the user.

**Comment by pde** [comment]
It seems that @johnridesabike's approach would optionally support @nhoizey's desire for sluggification too. Is there an elegant way to make that approach a switchable default?

Seems there could be two versions of this, one where that line needs to be explicitly present in `.eleventy.js`, and another where that handler is the default but can be overridden. Are there strong views on which is better?

**Comment by cokkto** [comment]
Also remeber about non-latin characters - the issue relates to this one. Tag transform function is really demanded, otherwise urls become the mess of latin and non latin letters and whitespaces

**Comment by Brixy** [comment]
I second this as I currently have problems transferring a website with blog to Eleventy from Wordpress because of an inconsistent usage of tags.

The company uses camel/dromedary case like

```
tags:
  - camelCase
```

so the ‘original’ tag form is important here:

```
<h1>Tagged “{{ tag }}”</h1>
```

For permalink `slugify` should transform tags to lowercase only IMHO. (I am using Eleventy 1 beta 8.)

```
permalink: /tags/{{ tag | slugify }}/
```

Also, a `customSlugify` function (with `toLowerCase()`) does not work with tags although it does with `Markdown-It`.

**Comment by zachleat** [comment]
New home base for this is at https://github.com/11ty/eleventy/issues/2462 and I’ll think we’ll probably land with an opt-in change per this comment https://github.com/11ty/eleventy/issues/2462#issuecomment-1170357117
